src/mem/old_api.cc

```diff
@@ -310,7 +281,25 @@ memFindBufSizeType(size_t net_size, size_t * gross_size)
     mem_type type;
     size_t size;
 
-    if (net_size <= 2 * 1024) {
+    if (net_size <= 32) {
+        type = MEM_32B_BUF;
+        size = 32;
+    } else if (net_size <= 64) {
+        type = MEM_64B_BUF;
+        size = 64;
+    } else if (net_size <= 128) {
+        type = MEM_128B_BUF;
+        size = 128;
+    } else if (net_size <= 256) {
+        type = MEM_256B_BUF;
+        size = 256;
+    } else if (net_size <= 512) {
+        type = MEM_512B_BUF;
+        size = 512;
+    } else if (net_size <= 1024) {
+        type = MEM_1K_BUF;
+        size = 1024;
+    } else if (net_size <= 2 * 1024) {
```

Comment:
I added "Also added a few new pools for 2KB and smaller buffers." to the PR description because these added pools did not exist before from memAllocBuf() callers point of view and because set of added pools is not equivalent to the string pools in the code below (that were used by now-converted memAllocString() callers, among others):

```C++
static const PoolMeta PoolAttrs[mem_str_pool_count] = {
    {"Short Strings", Mem::Allocator::RoundedSize(36)}, /* to fit rfc1123 and similar */
    {"Medium Strings", Mem::Allocator::RoundedSize(128)}, /* to fit most urls */
    {"Long Strings", Mem::Allocator::RoundedSize(512)},
    {"1KB Strings", Mem::Allocator::RoundedSize(1024)},
```

We are changing a lot more here than the API, and we should document that.

Please adjust PR description further as needed, of course.